src/quick-lint-js/parse.h

```diff
@@ -3172,10 +3173,21 @@ class parser {
         // let x = 3;
         initialize_variable:
         case token_type::equal: {
+          token equal_token = this->peek();
           expression *ast = this->parse_expression_remainder(
               variable,
               precedence{.commas = false, .in_operator = allow_in_operator});
           this->visit_binding_element(ast, v, declaration_kind);
+          bool is_assignment_not_allowed =
+              this->in_loop_statement_ &&
```

Comment:
Must fix: `this->in_loop_statement_` is the wrong thing to check. Your code produces a false positive for the following code:
```js
function of() {}
while (true) {
  const x = 3
  of();
}
```
```
hello.js:3:11: error: cannot assign to loop variable in for of/in loop [E173]
```